Unit test sometimes fails on Jenkins due to the concurrency error

<!--🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅

Hello there! 😄

To expedite issue processing please search open and closed issues before submitting a new one.
Existing issues often contain information about workarounds, resolution, or progress updates.

Please do not remove sections of this template when submitting your issue.  If the section is not relevant to your issue, leave the section but enter "n/a" for not applicable.

Note some sections are marked "REQUIRED" and must be provided.  Never mark required sections "n/a".

🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅-->


# 🐞 Bug Report

### Affected Services [**REQUIRED**]
<!-- Can you pin-point one or more EdgeX services (Data, Metadata, Command, etc...) as the source of the bug? -->
<!-- ✍️edit: --> The issue is located in: bootstrap unit test


### Is this a regression?

<!-- Did this behavior work correctly in the previous version? -->
<!-- ✍️--> Yes, it happens recently


### Description and Minimal Reproduction [**REQUIRED**]
Trigger the unit test.  I found it in this PR, but disappear after recheck
https://github.com/edgexfoundry/go-mod-bootstrap/pull/455


## 🔥 Exception or Error
<pre><code>
<!-- If the issue is accompanied by an exception or an error, please share it below: -->
<!-- ✍️-->
[2023-03-02T17:21:12.463Z] fatal error: concurrent map iteration and map write
[2023-03-02T17:21:12.463Z]
[2023-03-02T17:21:12.463Z] goroutine 44 [running]:
[2023-03-02T17:21:12.463Z] runtime.throw({0xad9184?, 0xc0003e9948?})
[2023-03-02T17:21:12.463Z] /usr/local/go/src/runtime/panic.go:992 +0x71 fp=0xc0000c6eb8 sp=0xc0000c6e88 pc=0x437e31
[2023-03-02T17:21:12.463Z] runtime.mapiternext(0xc000480820?)
[2023-03-02T17:21:12.463Z] /usr/local/go/src/runtime/map.go:871 +0x4eb fp=0xc0000c6f28 sp=0xc0000c6eb8 pc=0x411d2b
[2023-03-02T17:21:12.463Z] reflect.mapiternext(0x4d44af?)
[2023-03-02T17:21:12.463Z] /usr/local/go/src/runtime/map.go:1378 +0x19 fp=0xc0000c6f40 sp=0xc0000c6f28 pc=0x464a79
[2023-03-02T17:21:12.463Z] reflect.(*MapIter).Next(0xc000626000?)
[2023-03-02T17:21:12.463Z] /usr/local/go/src/reflect/value.go:1789 +0x7e fp=0xc0000c6f70 sp=0xc0000c6f40 pc=0x4d46de
[2023-03-02T17:21:12.463Z] internal/fmtsort.Sort({0xa011e0?, 0xc0000f0000?, 0x8?})
[2023-03-02T17:21:12.463Z] /usr/local/go/src/internal/fmtsort/sort.go:63 +0x159 fp=0xc0000c7030 sp=0xc0000c6f70 pc=0x4df639
[2023-03-02T17:21:12.463Z] fmt.(*pp).printValue(0xc000624000, {0xa011e0?, 0xc0000f0000?, 0x10842?}, 0x76, 0x2)
[2023-03-02T17:21:12.463Z] /usr/local/go/src/fmt/print.go:769 +0x445 fp=0xc0000c7218 sp=0xc0000c7030 pc=0x4e70c5
[2023-03-02T17:21:12.463Z] fmt.(*pp).printValue(0xc000624000, {0xa31fa0?, 0xc0000f0000?, 0x7f0231c8d338?}, 0x76, 0x1)
[2023-03-02T17:21:12.463Z] /usr/local/go/src/fmt/print.go:806 +0x124e fp=0xc0000c7400 sp=0xc0000c7218 pc=0x4e7ece
[2023-03-02T17:21:12.463Z] fmt.(*pp).printValue(0xc000624000, {0xa822c0?, 0xc0000f0000?, 0xc000480820?}, 0x76, 0x0)
[2023-03-02T17:21:12.463Z] /usr/local/go/src/fmt/print.go:876 +0xce5 fp=0xc0000c75e8 sp=0xc0000c7400 pc=0x4e7965
[2023-03-02T17:21:12.463Z] fmt.(*pp).printArg(0xc000624000, {0xa822c0?, 0xc0000f0000}, 0x76)
[2023-03-02T17:21:12.463Z] /usr/local/go/src/fmt/print.go:712 +0x74c fp=0xc0000c7688 sp=0xc0000c75e8 pc=0x4e6bec
[2023-03-02T17:21:12.463Z] fmt.(*pp).doPrintf(0xc000624000, {0xac5d2d, 0xd}, {0xc0000c7998?, 0x1, 0x1})
[2023-03-02T17:21:12.463Z] /usr/local/go/src/fmt/print.go:1122 +0x7de fp=0xc0000c7780 sp=0xc0000c7688 pc=0x4e993e
[2023-03-02T17:21:12.463Z] fmt.Sprintf({0xac5d2d, 0xd}, {0xc0000c7998, 0x1, 0x1})
[2023-03-02T17:21:12.463Z] /usr/local/go/src/fmt/print.go:219 +0x59 fp=0xc0000c77d8 sp=0xc0000c7780 pc=0x4e3859
[2023-03-02T17:21:12.463Z] github.com/stretchr/testify/mock.Arguments.Diff({0xc000433bc0, 0x2, 0x4?}, {0xc000540000, 0x2, 0x0?})
[2023-03-02T17:21:12.463Z] /go/pkg/mod/github.com/stretchr/testify@v1.8.2/mock/mock.go:889 +0x156 fp=0xc0000c7a60 sp=0xc0000c77d8 pc=0x966d36
[2023-03-02T17:21:12.463Z] github.com/stretchr/testify/mock.(*Mock).findExpectedCall(0xc0000c7b30?, {0xc94bec, 0x6}, {0xc000540000, 0x2, 0x2})
[2023-03-02T17:21:12.463Z] /go/pkg/mod/github.com/stretchr/testify@v1.8.2/mock/mock.go:361 +0xf3 fp=0xc0000c7ac8 sp=0xc0000c7a60 pc=0x963253
[2023-03-02T17:21:12.463Z] github.com/stretchr/testify/mock.(*Mock).MethodCalled(0xc0003b37c0, {0xc94bec, 0x6}, {0xc000540000, 0x2, 0x2})
[2023-03-02T17:21:12.463Z] /go/pkg/mod/github.com/stretchr/testify@v1.8.2/mock/mock.go:466 +0xbd fp=0xc0000c7d60 sp=0xc0000c7ac8 pc=0x963bbd
[2023-03-02T17:21:12.463Z] github.com/stretchr/testify/mock.(*Mock).Called(0x0?, {0xc000540000, 0x2, 0x2})
[2023-03-02T17:21:12.463Z] /go/pkg/mod/github.com/stretchr/testify@v1.8.2/mock/mock.go:456 +0x148 fp=0xc0000c7df8 sp=0xc0000c7d60 pc=0x963a88
[2023-03-02T17:21:12.463Z] github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/interfaces/mocks.(*MetricsReporter).Report(0xc0003b37c0, {0xbb1de8?, 0xc0000f0000}, 0xc00053e000)
[2023-03-02T17:21:12.463Z] /w/workspace/xfoundry_go-mod-bootstrap_PR-455/bootstrap/interfaces/mocks/MetricsReporter.go:17 +0xca fp=0xc0000c7e60 sp=0xc0000c7df8 pc=0x96b22a
[2023-03-02T17:21:12.463Z] github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/metrics.(*manager).Run.func1()
[2023-03-02T17:21:12.463Z] /w/workspace/xfoundry_go-mod-bootstrap_PR-455/bootstrap/metrics/manager.go:117 +0x1a5 fp=0xc0000c7fe0 sp=0xc0000c7e60 pc=0x96cc45
[2023-03-02T17:21:12.463Z] runtime.goexit()
[2023-03-02T17:21:12.463Z] /usr/local/go/src/runtime/asm_amd64.s:1571 +0x1 fp=0xc0000c7fe8 sp=0xc0000c7fe0 pc=0x46ae81
[2023-03-02T17:21:12.463Z] created by github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/metrics.(*manager).Run
[2023-03-02T17:21:12.463Z] /w/workspace/xfoundry_go-mod-bootstrap_PR-455/bootstrap/metrics/manager.go:105 +0x12a
[2023-03-02T17:21:12.463Z]
[2023-03-02T17:21:12.463Z] goroutine 1 [chan receive]:
[2023-03-02T17:21:12.463Z] testing.(*T).Run(0xc000329520, {0xad1a55?, 0x511fe5?}, 0xb08ab8)
[2023-03-02T17:21:12.463Z] /usr/local/go/src/testing/testing.go:1487 +0x37a
[2023-03-02T17:21:12.463Z] testing.runTests.func1(0xc000329520?)
[2023-03-02T17:21:12.463Z] /usr/local/go/src/testing/testing.go:1839 +0x6e
[2023-03-02T17:21:12.463Z] testing.tRunner(0xc000329520, 0xc00029fca0)
[2023-03-02T17:21:12.463Z] /usr/local/go/src/testing/testing.go:1439 +0x102
[2023-03-02T17:21:12.463Z] testing.runTests(0xc000438460?, {0xf83860, 0xa, 0xa}, {0x7f0231c845b8?, 0x40?, 0xf916c0?})
[2023-03-02T17:21:12.463Z] /usr/local/go/src/testing/testing.go:1837 +0x457
[2023-03-02T17:21:12.463Z] testing.(*M).Run(0xc000438460)
[2023-03-02T17:21:12.463Z] /usr/local/go/src/testing/testing.go:1719 +0x5d9
[2023-03-02T17:21:12.463Z] main.main()
[2023-03-02T17:21:12.463Z] _testmain.go:115 +0x255
[2023-03-02T17:21:12.463Z]
[2023-03-02T17:21:12.463Z] goroutine 59 [runnable]:
[2023-03-02T17:21:12.463Z] testing.(*T).Run(0xc00047a9c0, {0xace769?, 0xc00006be70?}, 0xc00048e8a0)
[2023-03-02T17:21:12.463Z] /usr/local/go/src/testing/testing.go:1464 +0x125
[2023-03-02T17:21:12.463Z] github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/metrics.TestMessageBusReporter_Report(0xc00047a9c0?)
[2023-03-02T17:21:12.463Z] /w/workspace/xfoundry_go-mod-bootstrap_PR-455/bootstrap/metrics/reporter_test.go:197 +0x12bc
[2023-03-02T17:21:12.463Z] testing.tRunner(0xc00047a9c0, 0xb08ab8)
[2023-03-02T17:21:12.463Z] /usr/local/go/src/testing/testing.go:1439 +0x102
[2023-03-02T17:21:12.463Z] created by testing.(*T).Run
[2023-03-02T17:21:12.463Z] /usr/local/go/src/testing/testing.go:1486 +0x35f
[2023-03-02T17:21:12.463Z]
[2023-03-02T17:21:12.463Z] goroutine 24 [chan receive]:
[2023-03-02T17:21:12.463Z] github.com/rcrowley/go-metrics.(*meterArbiter).tick(0xf91140)
[2023-03-02T17:21:12.463Z] /go/pkg/mod/github.com/rcrowley/go-metrics@v0.0.0-20201227073835-cf1acfcdf475/meter.go:239 +0x2a
[2023-03-02T17:21:12.463Z] created by github.com/rcrowley/go-metrics.NewMeter
[2023-03-02T17:21:12.463Z] /go/pkg/mod/github.com/rcrowley/go-metrics@v0.0.0-20201227073835-cf1acfcdf475/meter.go:46 +0xd3
[2023-03-02T17:21:12.463Z] FAIL github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/metrics 0.319s
</code></pre>

it happens again
```
[2023-03-25T04:23:24.198Z] fatal error: concurrent map iteration and map write

[2023-03-25T04:23:24.198Z] 

[2023-03-25T04:23:24.198Z] goroutine 72 [running]:

[2023-03-25T04:23:24.198Z] reflect.mapiterinit(0x451372?, 0x30?, 0xaf76a0?)

[2023-03-25T04:23:24.198Z] 	/usr/local/go/src/runtime/map.go:1375 +0x19

[2023-03-25T04:23:24.198Z] reflect.(*MapIter).Next(0xa3caa0?)

[2023-03-25T04:23:24.198Z] 	/usr/local/go/src/reflect/value.go:1919 +0x65

[2023-03-25T04:23:24.198Z] internal/fmtsort.Sort({0xa3caa0?, 0xc000080020?, 0xc00008003c?})

[2023-03-25T04:23:24.198Z] 	/usr/local/go/src/internal/fmtsort/sort.go:62 +0x1f0

[2023-03-25T04:23:24.198Z] fmt.(*pp).printValue(0xc0005c40d0, {0xa3caa0?, 0xc000080020?, 0x28?}, 0x76, 0x2)

[2023-03-25T04:23:24.198Z] 	/usr/local/go/src/fmt/print.go:816 +0x986

[2023-03-25T04:23:24.198Z] fmt.(*pp).printValue(0xc0005c40d0, {0xa6f920?, 0xc000080020?, 0x0?}, 0x76, 0x1)

[2023-03-25T04:23:24.198Z] 	/usr/local/go/src/fmt/print.go:853 +0x120a

[2023-03-25T04:23:24.198Z] fmt.(*pp).printValue(0xc0005c40d0, {0xac3200?, 0xc000080020?, 0xc0002705a0?}, 0x76, 0x0)

[2023-03-25T04:23:24.198Z] 	/usr/local/go/src/fmt/print.go:923 +0xaf0

[2023-03-25T04:23:24.198Z] fmt.(*pp).printArg(0xc0005c40d0, {0xac3200?, 0xc000080020}, 0x76)

[2023-03-25T04:23:24.198Z] 	/usr/local/go/src/fmt/print.go:759 +0x756

[2023-03-25T04:23:24.198Z] fmt.(*pp).doPrintf(0xc0005c40d0, {0xb07a8a, 0xd}, {0xc0004c99a0?, 0x1, 0x1})

[2023-03-25T04:23:24.198Z] 	/usr/local/go/src/fmt/print.go:1176 +0x9a9

[2023-03-25T04:23:24.198Z] fmt.Sprintf({0xb07a8a, 0xd}, {0xc0004c99a0, 0x1, 0x1})

[2023-03-25T04:23:24.198Z] 	/usr/local/go/src/fmt/print.go:239 +0x59

[2023-03-25T04:23:24.198Z] github.com/stretchr/testify/mock.Arguments.Diff({0xc0002794c0, 0x2, 0xc0004c9ac8?}, {0xc0004402a0, 0x2, 0xc0001c2c40?})

[2023-03-25T04:23:24.198Z] 	/go/pkg/mod/github.com/stretchr/testify@v1.8.2/mock/mock.go:897 +0x25e

[2023-03-25T04:23:24.198Z] github.com/stretchr/testify/mock.(*Mock).findExpectedCall(0x40fd28?, {0xcdbb50, 0x6}, {0xc0004402a0, 0x2, 0x2})

[2023-03-25T04:23:24.198Z] 	/go/pkg/mod/github.com/stretchr/testify@v1.8.2/mock/mock.go:361 +0xf3

[2023-03-25T04:23:24.198Z] github.com/stretchr/testify/mock.(*Mock).MethodCalled(0xc0003f85a0, {0xcdbb50, 0x6}, {0xc0004402a0, 0x2, 0x2})

[2023-03-25T04:23:24.198Z] 	/go/pkg/mod/github.com/stretchr/testify@v1.8.2/mock/mock.go:466 +0xbd

[2023-03-25T04:23:24.198Z] github.com/stretchr/testify/mock.(*Mock).Called(0x0?, {0xc0004402a0, 0x2, 0x2})

[2023-03-25T04:23:24.198Z] 	/go/pkg/mod/github.com/stretchr/testify@v1.8.2/mock/mock.go:456 +0x148

[2023-03-25T04:23:24.198Z] github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/interfaces/mocks.(*MetricsReporter).Report(0xc0003f85a0, {0xbee988?, 0xc000080020}, 0xc000446b10)

[2023-03-25T04:23:24.198Z] 	/w/workspace/xfoundry_go-mod-bootstrap_PR-489/bootstrap/interfaces/mocks/MetricsReporter.go:17 +0xbf

[2023-03-25T04:23:24.198Z] github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/metrics.(*manager).Run.func1()

[2023-03-25T04:23:24.198Z] 	/w/workspace/xfoundry_go-mod-bootstrap_PR-489/bootstrap/metrics/manager.go:117 +0x1bd

[2023-03-25T04:23:24.198Z] created by github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/metrics.(*manager).Run

[2023-03-25T04:23:24.198Z] 	/w/workspace/xfoundry_go-mod-bootstrap_PR-489/bootstrap/metrics/manager.go:105 +0x14a

[2023-03-25T04:23:24.198Z] 

[2023-03-25T04:23:24.198Z] goroutine 1 [chan receive]:

[2023-03-25T04:23:24.198Z] testing.(*T).Run(0xc0000cb6c0, {0xb13df4?, 0x561545?}, 0xb4cf88)

[2023-03-25T04:23:24.198Z] 	/usr/local/go/src/testing/testing.go:1630 +0x405

[2023-03-25T04:23:24.198Z] testing.runTests.func1(0x1000100?)

[2023-03-25T04:23:24.198Z] 	/usr/local/go/src/testing/testing.go:2036 +0x45

[2023-03-25T04:23:24.198Z] testing.tRunner(0xc0000cb6c0, 0xc00031fc80)

[2023-03-25T04:23:24.198Z] 	/usr/local/go/src/testing/testing.go:1576 +0x10b

[2023-03-25T04:23:24.198Z] testing.runTests(0xc000478320?, {0xff1940, 0xa, 0xa}, {0xc000471a68?, 0x100c00031fd08?, 0xfff760?})

[2023-03-25T04:23:24.198Z] 	/usr/local/go/src/testing/testing.go:2034 +0x489

[2023-03-25T04:23:24.198Z] testing.(*M).Run(0xc000478320)

[2023-03-25T04:23:24.198Z] 	/usr/local/go/src/testing/testing.go:1906 +0x63a

[2023-03-25T04:23:24.198Z] main.main()

[2023-03-25T04:23:24.198Z] 	_testmain.go:96 +0x1c5

[2023-03-25T04:23:24.198Z] 

[2023-03-25T04:23:24.198Z] goroutine 9 [chan receive]:

[2023-03-25T04:23:24.198Z] github.com/rcrowley/go-metrics.(*meterArbiter).tick(0xfff200)

[2023-03-25T04:23:24.198Z] 	/go/pkg/mod/github.com/rcrowley/go-metrics@v0.0.0-20201227073835-cf1acfcdf475/meter.go:239 +0x2a

[2023-03-25T04:23:24.198Z] created by github.com/rcrowley/go-metrics.NewMeter

[2023-03-25T04:23:24.198Z] 	/go/pkg/mod/github.com/rcrowley/go-metrics@v0.0.0-20201227073835-cf1acfcdf475/meter.go:46 +0xd3

[2023-03-25T04:23:24.198Z] 

[2023-03-25T04:23:24.198Z] goroutine 30 [chan receive]:

[2023-03-25T04:23:24.198Z] testing.(*T).Run(0xc000580ea0, {0xb0e43f?, 0x0?}, 0xc00010ede0)

[2023-03-25T04:23:24.198Z] 	/usr/local/go/src/testing/testing.go:1630 +0x405

[2023-03-25T04:23:24.198Z] github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/metrics.TestMessageBusReporter_Report(0xc000580ea0?)

[2023-03-25T04:23:24.198Z] 	/w/workspace/xfoundry_go-mod-bootstrap_PR-489/bootstrap/metrics/reporter_test.go:197 +0x1186

[2023-03-25T04:23:24.198Z] testing.tRunner(0xc000580ea0, 0xb4cf88)

[2023-03-25T04:23:24.198Z] 	/usr/local/go/src/testing/testing.go:1576 +0x10b

[2023-03-25T04:23:24.198Z] created by testing.(*T).Run

[2023-03-25T04:23:24.198Z] 	/usr/local/go/src/testing/testing.go:1629 +0x3ea

[2023-03-25T04:23:24.198Z] 

[2023-03-25T04:23:24.198Z] goroutine 99 [runnable]:

[2023-03-25T04:23:24.198Z] encoding/json.sliceEncoder.encode({0xa078e0?}, 0xc00001e900, {0x9f7d00?, 0xc00010eec0?, 0x4053f4?}, {0x9?, 0x0?})

[2023-03-25T04:23:24.198Z] 	/usr/local/go/src/encoding/json/encode.go:887 +0x32f

[2023-03-25T04:23:24.198Z] encoding/json.structEncoder.encode({{{0xc00047c480?, 0xc0005a3c50?, 0xc000071618?}, 0xc0005abf20?}}, 0xc00001e900, {0xab20a0?, 0xc00010eea0?, 0x20?}, {0x0, 0x1})

[2023-03-25T04:23:24.198Z] 	/usr/local/go/src/encoding/json/encode.go:759 +0x1f4

[2023-03-25T04:23:24.198Z] encoding/json.(*encodeState).reflectValue(0xc000071638?, {0xab20a0?, 0xc00010eea0?, 0xa6f880?}, {0xe0?, 0x60?})

[2023-03-25T04:23:24.198Z] 	/usr/local/go/src/encoding/json/encode.go:358 +0x78

[2023-03-25T04:23:24.198Z] encoding/json.(*encodeState).marshal(0xb037e9?, {0xab20a0?, 0xc00010eea0?}, {0x28?, 0xfd?})

[2023-03-25T04:23:24.198Z] 	/usr/local/go/src/encoding/json/encode.go:330 +0xfa

[2023-03-25T04:23:24.198Z] encoding/json.Marshal({0xab20a0, 0xc00010eea0})

[2023-03-25T04:23:24.198Z] 	/usr/local/go/src/encoding/json/encode.go:161 +0xe5

[2023-03-25T04:23:24.198Z] github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/metrics.(*messageBusReporter).Report.func1({0xb06dab, 0xb}, {0xac8ae0?, 0xc0005a8e10})

[2023-03-25T04:23:24.198Z] 	/w/workspace/xfoundry_go-mod-bootstrap_PR-489/bootstrap/metrics/reporter.go:174 +0x149b

[2023-03-25T04:23:24.198Z] github.com/rcrowley/go-metrics.(*StandardRegistry).Each(0x0?, 0xc00032a740)

[2023-03-25T04:23:24.198Z] 	/go/pkg/mod/github.com/rcrowley/go-metrics@v0.0.0-20201227073835-cf1acfcdf475/registry.go:70 +0x59

[2023-03-25T04:23:24.198Z] github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/metrics.(*messageBusReporter).Report(0xc000092a50, {0xbee988, 0xc000080020}, 0x0)

[2023-03-25T04:23:24.198Z] 	/w/workspace/xfoundry_go-mod-bootstrap_PR-489/bootstrap/metrics/reporter.go:106 +0x4d7

[2023-03-25T04:23:24.198Z] github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/metrics.TestMessageBusReporter_Report.func1(0xc000581860)

[2023-03-25T04:23:24.198Z] 	/w/workspace/xfoundry_go-mod-bootstrap_PR-489/bootstrap/metrics/reporter_test.go:229 +0x518

[2023-03-25T04:23:24.198Z] testing.tRunner(0xc000581860, 0xc00010ede0)

[2023-03-25T04:23:24.198Z] 	/usr/local/go/src/testing/testing.go:1576 +0x10b

[2023-03-25T04:23:24.198Z] created by testing.(*T).Run

[2023-03-25T04:23:24.198Z] 	/usr/local/go/src/testing/testing.go:1629 +0x3ea

[2023-03-25T04:23:24.198Z] FAIL	github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/metrics	0.324s

[2023-03-25T04:23:24.198Z] ok  	github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/registration	0.015s	coverage: 25.0% of statements

[2023-03-25T04:23:24.198Z] ok  	github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/secret	3.057s	coverage: 79.2% of statements

[2023-03-25T04:23:24.198Z] ok  	github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/utils	0.012s	coverage: 80.0% of statements

[2023-03-25T04:23:24.198Z] ok  	github.com/edgexfoundry/go-mod-bootstrap/v3/config	0.012s	coverage: 31.2% of statements

[2023-03-25T04:23:24.198Z] ok  	github.com/edgexfoundry/go-mod-bootstrap/v3/di	0.004s	coverage: 91.3% of statements

[2023-03-25T04:23:24.198Z] FAIL

[2023-03-25T04:23:24.198Z] make: *** [Makefile:10: unittest] Error 1
```

I have been able to reproduce this by calling running the `metrics` tests using `-test.count 10`, and running with debugger enabled on IntelliJ Ultimate 2023.1
```
GOROOT=/opt/go #gosetup
GOPATH=/home/anthony/go #gosetup
/opt/go/bin/go test -c -o /home/anthony/.cache/JetBrains/IntelliJIdea2023.1/tmp/GoLand/___1go_test_github_com_edgexfoundry_go_mod_bootstrap_v3_bootstrap_metrics.test -gcflags all=-N -l github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/metrics #gosetup
/opt/go/bin/go tool test2json -t /opt/apps/jetbrains/apps/IDEA-U/ch-0/231.8770.65.plugins/go-plugin/lib/dlv/linux/dlv --listen=127.0.0.1:43593 --headless=true --api-version=2 --check-go-version=false --only-same-user=false exec /home/anthony/.cache/JetBrains/IntelliJIdea2023.1/tmp/GoLand/___1go_test_github_com_edgexfoundry_go_mod_bootstrap_v3_bootstrap_metrics.test -- -test.v -test.paniconexit0 -test.count 10
API server listening at: 127.0.0.1:43593
=== RUN   TestNewManager
--- PASS: TestNewManager (0.00s)
=== RUN   TestNewManager
--- PASS: TestNewManager (0.00s)
=== RUN   TestNewManager
--- PASS: TestNewManager (0.00s)
=== RUN   TestNewManager
--- PASS: TestNewManager (0.00s)
=== RUN   TestNewManager
--- PASS: TestNewManager (0.00s)
=== RUN   TestMessageBusReporter_Register
--- PASS: TestMessageBusReporter_Register (0.00s)
=== RUN   TestMessageBusReporter_Register
--- PASS: TestMessageBusReporter_Register (0.00s)
=== RUN   TestMessageBusReporter_Register
--- PASS: TestMessageBusReporter_Register (0.00s)
=== RUN   TestMessageBusReporter_Register
--- PASS: TestMessageBusReporter_Register (0.00s)
=== RUN   TestMessageBusReporter_Register
--- PASS: TestMessageBusReporter_Register (0.00s)
=== RUN   TestManager_Get
--- PASS: TestManager_Get (0.00s)
=== RUN   TestManager_Get/Happy_path_Counter
    --- PASS: TestManager_Get/Happy_path_Counter (0.00s)
=== RUN   TestManager_Get/Happy_path_Counter
    --- PASS: TestManager_Get/Happy_path_Counter (0.00s)
=== RUN   TestManager_Get/Happy_path_Counter
    --- PASS: TestManager_Get/Happy_path_Counter (0.00s)
=== RUN   TestManager_Get/Happy_path_Counter
    --- PASS: TestManager_Get/Happy_path_Counter (0.00s)
=== RUN   TestManager_Get/Happy_path_Counter
=== RUN   TestManager_Get/Not_registered_Counter
    --- PASS: TestManager_Get/Not_registered_Counter (0.00s)
=== RUN   TestManager_Get/Not_registered_Counter
    --- PASS: TestManager_Get/Not_registered_Counter (0.00s)
=== RUN   TestManager_Get/Not_registered_Counter
    --- PASS: TestManager_Get/Not_registered_Counter (0.00s)
=== RUN   TestManager_Get/Not_registered_Counter
    --- PASS: TestManager_Get/Not_registered_Counter (0.00s)
=== RUN   TestManager_Get/Not_registered_Counter
=== RUN   TestManager_Get/Wrong_type_Counter
    manager_test.go:107: PASS:	Warnf(string,string)
    --- PASS: TestManager_Get/Wrong_type_Counter (0.00s)
=== RUN   TestManager_Get/Wrong_type_Counter
    manager_test.go:107: PASS:	Warnf(string,string)
    --- PASS: TestManager_Get/Wrong_type_Counter (0.00s)
=== RUN   TestManager_Get/Wrong_type_Counter
    manager_test.go:107: PASS:	Warnf(string,string)
    --- PASS: TestManager_Get/Wrong_type_Counter (0.00s)
=== RUN   TestManager_Get/Wrong_type_Counter
    manager_test.go:107: PASS:	Warnf(string,string)
    --- PASS: TestManager_Get/Wrong_type_Counter (0.00s)
=== RUN   TestManager_Get/Wrong_type_Counter
=== RUN   TestManager_Get/Happy_path_Gauge
    --- PASS: TestManager_Get/Happy_path_Gauge (0.00s)
=== RUN   TestManager_Get/Happy_path_Gauge
    --- PASS: TestManager_Get/Happy_path_Gauge (0.00s)
=== RUN   TestManager_Get/Happy_path_Gauge
    --- PASS: TestManager_Get/Happy_path_Gauge (0.00s)
=== RUN   TestManager_Get/Happy_path_Gauge
    --- PASS: TestManager_Get/Happy_path_Gauge (0.00s)
=== RUN   TestManager_Get/Not_registered_Gauge
    --- PASS: TestManager_Get/Not_registered_Gauge (0.00s)
=== RUN   TestManager_Get/Not_registered_Gauge
    --- PASS: TestManager_Get/Not_registered_Gauge (0.00s)
=== RUN   TestManager_Get/Not_registered_Gauge
    --- PASS: TestManager_Get/Not_registered_Gauge (0.00s)
=== RUN   TestManager_Get/Not_registered_Gauge
    --- PASS: TestManager_Get/Not_registered_Gauge (0.00s)
=== RUN   TestManager_Get/Wrong_type_Gauge
    manager_test.go:107: PASS:	Warnf(string,string)
    manager_test.go:107: PASS:	Warnf(string,string)
    --- PASS: TestManager_Get/Wrong_type_Gauge (0.00s)
=== RUN   TestManager_Get/Wrong_type_Gauge
    manager_test.go:107: PASS:	Warnf(string,string)
    manager_test.go:107: PASS:	Warnf(string,string)
    --- PASS: TestManager_Get/Wrong_type_Gauge (0.00s)
=== RUN   TestManager_Get/Wrong_type_Gauge
    manager_test.go:107: PASS:	Warnf(string,string)
    manager_test.go:107: PASS:	Warnf(string,string)
    --- PASS: TestManager_Get/Wrong_type_Gauge (0.00s)
=== RUN   TestManager_Get/Wrong_type_Gauge
    manager_test.go:107: PASS:	Warnf(string,string)
    manager_test.go:107: PASS:	Warnf(string,string)
    --- PASS: TestManager_Get/Wrong_type_Gauge (0.00s)
=== RUN   TestManager_Get/Happy_path_GaugeFloat64
    --- PASS: TestManager_Get/Happy_path_GaugeFloat64 (0.00s)
=== RUN   TestManager_Get/Happy_path_GaugeFloat64
    --- PASS: TestManager_Get/Happy_path_GaugeFloat64 (0.00s)
=== RUN   TestManager_Get/Happy_path_GaugeFloat64
    --- PASS: TestManager_Get/Happy_path_GaugeFloat64 (0.00s)
=== RUN   TestManager_Get/Happy_path_GaugeFloat64
    --- PASS: TestManager_Get/Happy_path_GaugeFloat64 (0.00s)
=== RUN   TestManager_Get/Not_registered_GaugeFloat64
    --- PASS: TestManager_Get/Not_registered_GaugeFloat64 (0.00s)
=== RUN   TestManager_Get/Not_registered_GaugeFloat64
    --- PASS: TestManager_Get/Not_registered_GaugeFloat64 (0.00s)
=== RUN   TestManager_Get/Not_registered_GaugeFloat64
    --- PASS: TestManager_Get/Not_registered_GaugeFloat64 (0.00s)
=== RUN   TestManager_Get/Not_registered_GaugeFloat64
    --- PASS: TestManager_Get/Not_registered_GaugeFloat64 (0.00s)
=== RUN   TestManager_Get/Wrong_type_GaugeFloat64
    manager_test.go:107: PASS:	Warnf(string,string)
    manager_test.go:107: PASS:	Warnf(string,string)
    manager_test.go:107: PASS:	Warnf(string,string)
    --- PASS: TestManager_Get/Wrong_type_GaugeFloat64 (0.00s)
=== RUN   TestManager_Get/Wrong_type_GaugeFloat64
    manager_test.go:107: PASS:	Warnf(string,string)
    manager_test.go:107: PASS:	Warnf(string,string)
    manager_test.go:107: PASS:	Warnf(string,string)
    --- PASS: TestManager_Get/Wrong_type_GaugeFloat64 (0.00s)
=== RUN   TestManager_Get/Wrong_type_GaugeFloat64
    manager_test.go:107: PASS:	Warnf(string,string)
    manager_test.go:107: PASS:	Warnf(string,string)
    manager_test.go:107: PASS:	Warnf(string,string)
    --- PASS: TestManager_Get/Wrong_type_GaugeFloat64 (0.00s)
=== RUN   TestManager_Get/Wrong_type_GaugeFloat64
    manager_test.go:107: PASS:	Warnf(string,string)
    manager_test.go:107: PASS:	Warnf(string,string)
    manager_test.go:107: PASS:	Warnf(string,string)
    --- PASS: TestManager_Get/Wrong_type_GaugeFloat64 (0.00s)
=== RUN   TestManager_Get/Happy_path_Timer
    --- PASS: TestManager_Get/Happy_path_Timer (0.00s)
=== RUN   TestManager_Get/Happy_path_Timer
    --- PASS: TestManager_Get/Happy_path_Timer (0.00s)
=== RUN   TestManager_Get/Happy_path_Timer
    --- PASS: TestManager_Get/Happy_path_Timer (0.00s)
=== RUN   TestManager_Get/Happy_path_Timer
    --- PASS: TestManager_Get/Happy_path_Timer (0.00s)
=== RUN   TestManager_Get/Not_registered_Timer
    --- PASS: TestManager_Get/Not_registered_Timer (0.00s)
=== RUN   TestManager_Get/Not_registered_Timer
    --- PASS: TestManager_Get/Not_registered_Timer (0.00s)
=== RUN   TestManager_Get/Not_registered_Timer
    --- PASS: TestManager_Get/Not_registered_Timer (0.00s)
=== RUN   TestManager_Get/Not_registered_Timer
    --- PASS: TestManager_Get/Not_registered_Timer (0.00s)
=== RUN   TestManager_Get/Wrong_type_Timer
    manager_test.go:107: PASS:	Warnf(string,string)
    manager_test.go:107: PASS:	Warnf(string,string)
    manager_test.go:107: PASS:	Warnf(string,string)
    manager_test.go:107: PASS:	Warnf(string,string)
    --- PASS: TestManager_Get/Wrong_type_Timer (0.00s)
=== RUN   TestManager_Get/Wrong_type_Timer
    manager_test.go:107: PASS:	Warnf(string,string)
    manager_test.go:107: PASS:	Warnf(string,string)
    manager_test.go:107: PASS:	Warnf(string,string)
    manager_test.go:107: PASS:	Warnf(string,string)
    --- PASS: TestManager_Get/Wrong_type_Timer (0.00s)
=== RUN   TestManager_Get/Wrong_type_Timer
    manager_test.go:107: PASS:	Warnf(string,string)
    manager_test.go:107: PASS:	Warnf(string,string)
    manager_test.go:107: PASS:	Warnf(string,string)
    manager_test.go:107: PASS:	Warnf(string,string)
    --- PASS: TestManager_Get/Wrong_type_Timer (0.00s)
=== RUN   TestManager_Get/Wrong_type_Timer
    manager_test.go:107: PASS:	Warnf(string,string)
    manager_test.go:107: PASS:	Warnf(string,string)
    manager_test.go:107: PASS:	Warnf(string,string)
    manager_test.go:107: PASS:	Warnf(string,string)
    --- PASS: TestManager_Get/Wrong_type_Timer (0.00s)
=== RUN   TestManager_Get
--- PASS: TestManager_Get (0.00s)
=== RUN   TestManager_Get
--- PASS: TestManager_Get (0.00s)
=== RUN   TestManager_Get
--- PASS: TestManager_Get (0.00s)
=== RUN   TestManager_Get
=== RUN   TestManager_Register_Unregister
--- PASS: TestManager_Register_Unregister (0.00s)
=== RUN   TestManager_Register_Unregister
    manager_test.go:121: 
        	Error Trace:	/opt/intel/projects/go-mod-bootstrap/bootstrap/metrics/manager_test.go:121
        	Error:      	Received unexpected error:
        	            	duplicate metric: my-counter
        	Test:       	TestManager_Register_Unregister
--- FAIL: TestManager_Register_Unregister (0.00s)
=== RUN   TestManager_Register_Unregister
    manager_test.go:121: 
        	Error Trace:	/opt/intel/projects/go-mod-bootstrap/bootstrap/metrics/manager_test.go:121
        	Error:      	Received unexpected error:
        	            	duplicate metric: my-counter
        	Test:       	TestManager_Register_Unregister
--- FAIL: TestManager_Register_Unregister (0.00s)
=== RUN   TestManager_Register_Unregister
    manager_test.go:121: 
        	Error Trace:	/opt/intel/projects/go-mod-bootstrap/bootstrap/metrics/manager_test.go:121
        	Error:      	Received unexpected error:
        	            	duplicate metric: my-counter
        	Test:       	TestManager_Register_Unregister
--- FAIL: TestManager_Register_Unregister (0.00s)



=== RUN   TestManager_Register_Error
--- PASS: TestManager_Register_Error (0.00s)
=== RUN   TestManager_Register_Error
    manager_test.go:144: 
        	Error Trace:	/opt/intel/projects/go-mod-bootstrap/bootstrap/metrics/manager_test.go:144
        	Error:      	Received unexpected error:
        	            	duplicate metric: my-counter
        	Test:       	TestManager_Register_Error
--- FAIL: TestManager_Register_Error (0.00s)
=== RUN   TestManager_Register_Error
    manager_test.go:144: 
        	Error Trace:	/opt/intel/projects/go-mod-bootstrap/bootstrap/metrics/manager_test.go:144
        	Error:      	Received unexpected error:
        	            	duplicate metric: my-counter
        	Test:       	TestManager_Register_Error
--- FAIL: TestManager_Register_Error (0.00s)
=== RUN   TestManager_Register_Error
    manager_test.go:144: 
        	Error Trace:	/opt/intel/projects/go-mod-bootstrap/bootstrap/metrics/manager_test.go:144
        	Error:      	Received unexpected error:
        	            	duplicate metric: my-counter
        	Test:       	TestManager_Register_Error
--- FAIL: TestManager_Register_Error (0.00s)



=== RUN   TestManager_Run
    manager_test.go:162: PASS:	Report(*metrics.StandardRegistry,map[string]map[string]string)
--- PASS: TestManager_Run (0.20s)
=== RUN   TestManager_Run
    manager_test.go:162: PASS:	Report(*metrics.StandardRegistry,map[string]map[string]string)
--- PASS: TestManager_Run (0.20s)
=== RUN   TestManager_Run
    manager_test.go:162: PASS:	Report(*metrics.StandardRegistry,map[string]map[string]string)
--- PASS: TestManager_Run (0.20s)
=== RUN   TestManager_Run
    manager_test.go:162: PASS:	Report(*metrics.StandardRegistry,map[string]map[string]string)
--- PASS: TestManager_Run (0.20s)
=== RUN   TestManager_Run_Error
    manager_test.go:188: PASS:	Report(*metrics.StandardRegistry,map[string]map[string]string)
    manager_test.go:189: PASS:	Errorf(string,string)
    manager_test.go:189: PASS:	Infof(string,string)
--- PASS: TestManager_Run_Error (0.10s)
=== RUN   TestManager_Run_Error
    manager_test.go:188: PASS:	Report(*metrics.StandardRegistry,map[string]map[string]string)
    manager_test.go:189: PASS:	Errorf(string,string)
    manager_test.go:189: PASS:	Infof(string,string)
--- PASS: TestManager_Run_Error (0.10s)
=== RUN   TestManager_Run_Error
    manager_test.go:188: PASS:	Report(*metrics.StandardRegistry,map[string]map[string]string)
    manager_test.go:189: PASS:	Errorf(string,string)
    manager_test.go:189: PASS:	Infof(string,string)
--- PASS: TestManager_Run_Error (0.10s)
=== RUN   TestManager_Run_Error
    manager_test.go:188: PASS:	Report(*metrics.StandardRegistry,map[string]map[string]string)
    manager_test.go:189: PASS:	Errorf(string,string)
    manager_test.go:189: PASS:	Infof(string,string)
--- PASS: TestManager_Run_Error (0.10s)
=== RUN   TestManager_ResetInterval
--- PASS: TestManager_ResetInterval (0.00s)
=== RUN   TestManager_ResetInterval
--- PASS: TestManager_ResetInterval (0.00s)
=== RUN   TestManager_ResetInterval
--- PASS: TestManager_ResetInterval (0.00s)
=== RUN   TestManager_ResetInterval
--- PASS: TestManager_ResetInterval (0.00s)
=== RUN   TestNewMessageBusReporter
--- PASS: TestNewMessageBusReporter (0.00s)
=== RUN   TestNewMessageBusReporter/Happy_path_no_additional_tags
    --- PASS: TestNewMessageBusReporter/Happy_path_no_additional_tags (0.00s)
=== RUN   TestNewMessageBusReporter/Happy_path_no_additional_tags
    --- PASS: TestNewMessageBusReporter/Happy_path_no_additional_tags (0.00s)
=== RUN   TestNewMessageBusReporter/Happy_path_no_additional_tags
    --- PASS: TestNewMessageBusReporter/Happy_path_no_additional_tags (0.00s)
=== RUN   TestNewMessageBusReporter/Happy_path_no_additional_tags
    --- PASS: TestNewMessageBusReporter/Happy_path_no_additional_tags (0.00s)
=== RUN   TestNewMessageBusReporter/Happy_path_with_additional_tags
    --- PASS: TestNewMessageBusReporter/Happy_path_with_additional_tags (0.00s)
=== RUN   TestNewMessageBusReporter/Happy_path_with_additional_tags
    --- PASS: TestNewMessageBusReporter/Happy_path_with_additional_tags (0.00s)
=== RUN   TestNewMessageBusReporter/Happy_path_with_additional_tags
    --- PASS: TestNewMessageBusReporter/Happy_path_with_additional_tags (0.00s)
=== RUN   TestNewMessageBusReporter/Happy_path_with_additional_tags
    --- PASS: TestNewMessageBusReporter/Happy_path_with_additional_tags (0.00s)
=== RUN   TestNewMessageBusReporter
--- PASS: TestNewMessageBusReporter (0.00s)
=== RUN   TestNewMessageBusReporter
--- PASS: TestNewMessageBusReporter (0.00s)
=== RUN   TestNewMessageBusReporter
--- PASS: TestNewMessageBusReporter (0.00s)
=== RUN   TestMessageBusReporter_Report
--- PASS: TestMessageBusReporter_Report (0.01s)
=== RUN   TestMessageBusReporter_Report/Happy_path_-_Counter
    --- PASS: TestMessageBusReporter_Report/Happy_path_-_Counter (0.00s)
=== RUN   TestMessageBusReporter_Report/Happy_path_-_Gauge
    --- PASS: TestMessageBusReporter_Report/Happy_path_-_Gauge (0.00s)
=== RUN   TestMessageBusReporter_Report/Happy_path_-_GaugeFloat64
    --- PASS: TestMessageBusReporter_Report/Happy_path_-_GaugeFloat64 (0.00s)
=== RUN   TestMessageBusReporter_Report/Happy_path_-_Timer
    --- PASS: TestMessageBusReporter_Report/Happy_path_-_Timer (0.00s)
=== RUN   TestMessageBusReporter_Report/Happy_path_-_Histogram
    --- PASS: TestMessageBusReporter_Report/Happy_path_-_Histogram (0.00s)
=== RUN   TestMessageBusReporter_Report/No_Metrics
    --- PASS: TestMessageBusReporter_Report/No_Metrics (0.00s)
=== RUN   TestMessageBusReporter_Report/Unsupported_Metric
    --- PASS: TestMessageBusReporter_Report/Unsupported_Metric (0.00s)
=== RUN   TestMessageBusReporter_Report
    reporter_test.go:124: 
        	Error Trace:	/opt/intel/projects/go-mod-bootstrap/bootstrap/metrics/reporter_test.go:124
        	Error:      	Received unexpected error:
        	            	duplicate metric: disabled-metric
        	Test:       	TestMessageBusReporter_Report
--- FAIL: TestMessageBusReporter_Report (0.00s)
=== RUN   TestMessageBusReporter_Report
    reporter_test.go:124: 
        	Error Trace:	/opt/intel/projects/go-mod-bootstrap/bootstrap/metrics/reporter_test.go:124
        	Error:      	Received unexpected error:
        	            	duplicate metric: disabled-metric
        	Test:       	TestMessageBusReporter_Report
--- FAIL: TestMessageBusReporter_Report (0.00s)
=== RUN   TestMessageBusReporter_Report
    reporter_test.go:124: 
        	Error Trace:	/opt/intel/projects/go-mod-bootstrap/bootstrap/metrics/reporter_test.go:124
        	Error:      	Received unexpected error:
        	            	duplicate metric: disabled-metric
        	Test:       	TestMessageBusReporter_Report
--- FAIL: TestMessageBusReporter_Report (0.00s)



fatal error: concurrent map iteration and map write

goroutine 105 [running]:
reflect.mapiterinit(0xd69800?, 0xc000158000?, 0x1b5?)
	/opt/go/src/runtime/map.go:1375 +0x19
reflect.(*MapIter).Next(0xc00073e980)
	/opt/go/src/reflect/value.go:1919 +0x132
internal/fmtsort.Sort({0xd69800, 0xc000158000, 0x1b5})
	/opt/go/src/internal/fmtsort/sort.go:62 +0x165
fmt.(*pp).printValue(0xc000746270, {0xd69800, 0xc000158000, 0x1b5}, 0x76, 0x2)
	/opt/go/src/fmt/print.go:816 +0x8a5
fmt.(*pp).printValue(0xc000746270, {0xd9cea0, 0xc000158000, 0x199}, 0x76, 0x1)
	/opt/go/src/fmt/print.go:853 +0x121a
fmt.(*pp).printValue(0xc000746270, {0xdef780, 0xc000158000, 0x16}, 0x76, 0x0)
	/opt/go/src/fmt/print.go:923 +0xa47
fmt.(*pp).printArg(0xc000746270, {0xdef780, 0xc000158000}, 0x76)
	/opt/go/src/fmt/print.go:759 +0xcb8
fmt.(*pp).doPrintf(0xc000746270, {0xe34b11, 0xd}, {0xc000835578, 0x1, 0x1})
	/opt/go/src/fmt/print.go:1176 +0x1136
fmt.Sprintf({0xe34b11, 0xd}, {0xc000835578, 0x1, 0x1})
	/opt/go/src/fmt/print.go:239 +0x7d
github.com/stretchr/testify/mock.Arguments.Diff({0xc0005a1c00, 0x2, 0x2}, {0xc000d12460, 0x2, 0x2})
	/home/anthony/go/pkg/mod/github.com/stretchr/testify@v1.8.2/mock/mock.go:897 +0x326
github.com/stretchr/testify/mock.(*Mock).findExpectedCall(0xc000632b90, {0xfb35b2, 0x6}, {0xc000d12460, 0x2, 0x2})
	/home/anthony/go/pkg/mod/github.com/stretchr/testify@v1.8.2/mock/mock.go:361 +0x17a
github.com/stretchr/testify/mock.(*Mock).MethodCalled(0xc000632b90, {0xfb35b2, 0x6}, {0xc000d12460, 0x2, 0x2})
	/home/anthony/go/pkg/mod/github.com/stretchr/testify@v1.8.2/mock/mock.go:466 +0xca
github.com/stretchr/testify/mock.(*Mock).Called(0xc000632b90, {0xc000d12460, 0x2, 0x2})
	/home/anthony/go/pkg/mod/github.com/stretchr/testify@v1.8.2/mock/mock.go:456 +0x20d
github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/interfaces/mocks.(*MetricsReporter).Report(0xc000632b90, {0xec5428, 0xc000158000}, 0xc0008b4390)
	/opt/intel/projects/go-mod-bootstrap/bootstrap/interfaces/mocks/MetricsReporter.go:17 +0x15c
github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/metrics.(*manager).Run.func1()
	/opt/intel/projects/go-mod-bootstrap/bootstrap/metrics/manager.go:117 +0x174
created by github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/metrics.(*manager).Run
	/opt/intel/projects/go-mod-bootstrap/bootstrap/metrics/manager.go:105 +0x199

goroutine 1 [chan receive]:
testing.(*T).Run(0xc000102ea0, {0xe36442, 0xf}, 0xe79160)
	/opt/go/src/testing/testing.go:1630 +0x7cb
testing.runTests.func1(0xc000602820)
	/opt/go/src/testing/testing.go:2036 +0xba
testing.tRunner(0xc000602820, 0xc00012b980)
	/opt/go/src/testing/testing.go:1576 +0x1ca
testing.runTests(0xc000012048, {0x12845c0, 0xa, 0xa}, {0x0, 0x0, 0x0})
	/opt/go/src/testing/testing.go:2034 +0x47d
testing.(*M).Run(0xc000424500)
	/opt/go/src/testing/testing.go:1906 +0xb8e
main.main()
	_testmain.go:65 +0xb7

goroutine 27 [chan receive, 1 minutes]:
github.com/rcrowley/go-metrics.(*meterArbiter).tick(0x1291e60)
	/home/anthony/go/pkg/mod/github.com/rcrowley/go-metrics@v0.0.0-20201227073835-cf1acfcdf475/meter.go:239 +0x39
created by github.com/rcrowley/go-metrics.NewMeter
	/home/anthony/go/pkg/mod/github.com/rcrowley/go-metrics@v0.0.0-20201227073835-cf1acfcdf475/meter.go:46 +0x191

goroutine 67 [runnable]:
regexp.(*inputs).init(0xc000e123d8, {0x0, 0x0}, {0x0, 0x0, 0x0}, {0xfb3558, 0x60})
	/opt/go/src/regexp/exec.go:85 +0x18c
regexp.(*Regexp).backtrack(0xc000a5c960, {0x0, 0x0, 0x0}, {0xfb3558, 0x60}, 0x0, 0x0, {0x12c1360, 0x0, ...})
	/opt/go/src/regexp/backtrack.go:318 +0x185
regexp.(*Regexp).doExecute(0xc000a5c960, {0x0, 0x0}, {0x0, 0x0, 0x0}, {0xfb3558, 0x60}, 0x0, 0x0, ...)
	/opt/go/src/regexp/exec.go:535 +0x2e5
regexp.(*Regexp).doMatch(0xc000a5c960, {0x0, 0x0}, {0x0, 0x0, 0x0}, {0xfb3558, 0x60})
	/opt/go/src/regexp/exec.go:514 +0xbe
regexp.(*Regexp).MatchString(0xc000a5c960, {0xfb3558, 0x60})
	/opt/go/src/regexp/regexp.go:533 +0x47
github.com/stretchr/testify/mock.(*Mock).Called(0xc000116230, {0xc000b843e0, 0x2, 0x2})
	/home/anthony/go/pkg/mod/github.com/stretchr/testify@v1.8.2/mock/mock.go:451 +0x128
github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/interfaces/mocks.(*MetricsReporter).Report(0xc000116230, {0xec5428, 0xc000158000}, 0xc0003d1ec0)
	/opt/intel/projects/go-mod-bootstrap/bootstrap/interfaces/mocks/MetricsReporter.go:17 +0x15c
github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/metrics.(*manager).Run.func1()
	/opt/intel/projects/go-mod-bootstrap/bootstrap/metrics/manager.go:117 +0x174
created by github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/metrics.(*manager).Run
	/opt/intel/projects/go-mod-bootstrap/bootstrap/metrics/manager.go:105 +0x199

goroutine 166 [runnable]:
github.com/stretchr/testify/assert.isTest({0xfad7ef, 0xa}, {0xe31adb, 0x9})
	/home/anthony/go/pkg/mod/github.com/stretchr/testify@v1.8.2/assert/assertions.go:169 +0x50
github.com/stretchr/testify/assert.CallerInfo()
	/home/anthony/go/pkg/mod/github.com/stretchr/testify@v1.8.2/assert/assertions.go:155 +0xcd6
github.com/stretchr/testify/mock.(*Mock).MethodCalled(0xc000890370, {0xfb2e23, 0x6}, {0xc000954810, 0x1, 0x1})
	/home/anthony/go/pkg/mod/github.com/stretchr/testify@v1.8.2/mock/mock.go:522 +0xd59
github.com/stretchr/testify/mock.(*Mock).Called(0xc000890370, {0xc000954810, 0x1, 0x1})
	/home/anthony/go/pkg/mod/github.com/stretchr/testify@v1.8.2/mock/mock.go:456 +0x20d
github.com/edgexfoundry/go-mod-core-contracts/v3/clients/logger/mocks.(*LoggingClient).Errorf(0xc000890370, {0xe2ecb6, 0x6}, {0x0, 0x0, 0x0})
	/home/anthony/go/pkg/mod/github.com/edgexfoundry/go-mod-core-contracts/v3@v3.0.0-dev.41/clients/logger/mocks/LoggingClient.go:45 +0x229
github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/metrics.(*manager).Run.func1()
	/opt/intel/projects/go-mod-bootstrap/bootstrap/metrics/manager.go:118 +0x1d2
created by github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/metrics.(*manager).Run
	/opt/intel/projects/go-mod-bootstrap/bootstrap/metrics/manager.go:105 +0x199

goroutine 164 [select]:
github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/metrics.(*manager).Run.func1()
	/opt/intel/projects/go-mod-bootstrap/bootstrap/metrics/manager.go:107 +0xdc
created by github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/metrics.(*manager).Run
	/opt/intel/projects/go-mod-bootstrap/bootstrap/metrics/manager.go:105 +0x199

goroutine 68 [chan receive]:
testing.(*T).Run(0xc000683ba0, {0xe38805, 0x12}, 0xc000666500)
	/opt/go/src/testing/testing.go:1630 +0x7cb
github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/metrics.TestManager_Get(0xc000102ea0)
	/opt/intel/projects/go-mod-bootstrap/bootstrap/metrics/manager_test.go:78 +0x1348
testing.tRunner(0xc000102ea0, 0xe79160)
	/opt/go/src/testing/testing.go:1576 +0x1ca
created by testing.(*T).Run
	/opt/go/src/testing/testing.go:1629 +0x7a6

goroutine 228 [runnable]:
regexp/syntax.(*compiler).compile(0xc00012f6b8, 0xc000420700)
	/opt/go/src/regexp/syntax/compile.go:105 +0x11ec
regexp/syntax.(*compiler).compile(0xc00012f6b8, 0xc000420a80)
	/opt/go/src/regexp/syntax/compile.go:145 +0xf53
regexp/syntax.Compile(0xc000420a80)
	/opt/go/src/regexp/syntax/compile.go:74 +0x4e
regexp.compile({0xe30f94, 0x8}, 0xd4, 0x0)
	/opt/go/src/regexp/regexp.go:182 +0x1ab
regexp.Compile({0xe30f94, 0x8})
	/opt/go/src/regexp/regexp.go:137 +0x56
regexp.MustCompile({0xe30f94, 0x8})
	/opt/go/src/regexp/regexp.go:317 +0x66
github.com/stretchr/testify/mock.(*Mock).Called(0xc000974230, {0xc0003a62e0, 0x2, 0x2})
	/home/anthony/go/pkg/mod/github.com/stretchr/testify@v1.8.2/mock/mock.go:450 +0x114
github.com/edgexfoundry/go-mod-core-contracts/v3/clients/logger/mocks.(*LoggingClient).Warnf(0xc000974230, {0xe57b07, 0x5a}, {0xc0003ac340, 0x1, 0x1})
	/home/anthony/go/pkg/mod/github.com/edgexfoundry/go-mod-core-contracts/v3@v3.0.0-dev.41/clients/logger/mocks/LoggingClient.go:123 +0x229
github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/metrics.(*manager).GetCounter(0xc000974280, {0xe32273, 0x9})
	/opt/intel/projects/go-mod-bootstrap/bootstrap/metrics/manager.go:158 +0x1df
github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/metrics.TestManager_Get.func1(0xc000683ba0)
	/opt/intel/projects/go-mod-bootstrap/bootstrap/metrics/manager_test.go:94 +0x3f2
testing.tRunner(0xc000683ba0, 0xc000666500)
	/opt/go/src/testing/testing.go:1576 +0x1ca
created by testing.(*T).Run
	/opt/go/src/testing/testing.go:1629 +0x7a6

Debugger finished with the exit code 0
```

--- a/bootstrap/metrics/manager.go
+++ b/bootstrap/metrics/manager.go
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright 2022 Intel Corp.
+ * Copyright 2022-2023 Intel Corp.
  *
  * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
  * in compliance with the License. You may obtain a copy of the License at
@@ -51,7 +51,7 @@ func (m *manager) ResetInterval(interval time.Duration) {
 func NewManager(lc logger.LoggingClient, interval time.Duration, reporter interfaces.MetricsReporter) interfaces.MetricsManager {
 	m := &manager{
 		lc:         lc,
-		registry:   gometrics.DefaultRegistry,
+		registry:   gometrics.NewRegistry(),
 		reporter:   reporter,
 		interval:   interval,
 		metricTags: make(map[string]map[string]string),
@@ -100,9 +100,10 @@ func (m *manager) Run(ctx context.Context, wg *sync.WaitGroup) {
 	m.ticker = time.NewTicker(m.interval)
 
 	wg.Add(1)
-	defer wg.Done()
 
 	go func() {
+		defer wg.Done()
+
 		for {
 			select {
 			case <-ctx.Done():

--- a/bootstrap/metrics/reporter_test.go
+++ b/bootstrap/metrics/reporter_test.go
@@ -113,7 +113,7 @@ func TestMessageBusReporter_Report(t *testing.T) {
 		expectedTags)
 	require.NoError(t, err)
 
-	reg := gometrics.DefaultRegistry
+	reg := gometrics.NewRegistry()
 
 	counter := gometrics.NewCounter()
 	counter.Inc(intValue)

